houston/configuration.py

```diff
@@ -111,13 +111,19 @@ def __getitem__(self, name: str) -> Any:
             msg = "no option [{}] in state [{}]"
             msg.format(name, self.__class__.__name__)
             raise KeyError(msg)
-        return getattr(self, var._field)
+        return getattr(self, opt._field)
+
+    def __hash__(self) -> int:
+        all_opts = [self[opt.name] for opt in self.__class__.options]
+        all_opts.insert(0, self.__class__.__name__)
+        return hash(tuple(all_opts))
 
     def __eq__(self, other: 'Configuration') -> bool:
         if type(self) != type(other):
-            msg = "illegal comparison of configurations: [{}] vs. [{}]"
-            msg = msg.format(self.__class__.__name__, state.__class__.__name__)
-            raise Exception(msg)  # FIXME use HoustonException
+            msg = "comparison of different classesof configs: [{}] vs. [{}]"
+            msg = msg.format(self.__class__.__name__, other.__class__.__name__)
+            logger.warning(msg)
```

Comment:
Raise an actual warning via `warnings.warn`. The logger will automatically capture it.